geode-core/src/main/java/org/apache/geode/internal/cache/TXRegionState.java

```diff
@@ -335,16 +345,26 @@ void buildMessage(InternalRegion r, TXCommitMessage msg) {
           TXEntryState txes = (TXEntryState) me.getValue();
           txes.buildMessage(r, eKey, msg, this.otherMembers);
           if (txes.getFilterRoutingInfo() != null) {
+            Set<InternalDistributedMember> tempSet = new HashSet();
+            if (txes.getAdjunctRecipients() != null) {
+              tempSet.addAll(txes.getAdjunctRecipients());
+            }
+            // exclude members that actually host targeted bucket from notification only list
+            tempSet.removeAll(redundantMemberSet);
+            if (!tempSet.isEmpty()) {
+              if (msg.getNotificationOnlyMembers().get(regionCommit) == null) {
+                msg.getNotificationOnlyMembers().put(regionCommit, tempSet);
+              } else {
+                msg.getNotificationOnlyMembers().get(regionCommit).addAll(tempSet);
+              }
+            }
```

Comment:
If the value returned from `txes.getAdjunctRecipients()` is null, then we know that `tempSet` will be empty, so we can skip the subsequent `removeAll()` and `isEmpty()` checks, because we know the outcome of them. With that in mind, this block can be refactored to:
```
            if (txes.getAdjunctRecipients() != null) {
              Set<InternalDistributedMember> tempSet = new HashSet<>(txes.getAdjunctRecipients());
              // exclude members that actually host targeted bucket from notification only list
              tempSet.removeAll(redundantMemberSet);
              if (!tempSet.isEmpty()) {
                if (msg.getNotificationOnlyMembers().get(regionCommit) == null) {
                  msg.getNotificationOnlyMembers().put(regionCommit, tempSet);
                } else {
                  msg.getNotificationOnlyMembers().get(regionCommit).addAll(tempSet);
                }
              }
            }
            newMemberSet.addAll(txes.getFilterRoutingInfo().getMembers());
```